sdk/storage/azblob/blob/client.go

```diff
@@ -228,10 +229,10 @@ func (b *Client) CopyFromURL(ctx context.Context, copySource string, options *Co
 
 // GetSASToken is a convenience method for generating a SAS token for the currently pointed at blob.
 // It can only be used if the credential supplied during creation was a SharedKeyCredential.
-func (b *Client) GetSASToken(permissions SASPermissions, start time.Time, expiry time.Time) (string, error) {
-	urlParts, _ := exported.ParseURL(b.URL())
+func (b *Client) GetSASToken(permissions sas.BlobPermissions, start time.Time, expiry time.Time) (string, error) {
+	urlParts, _ := ParseURL(b.URL())
```

Comment:
I thought we were putting this function in the tiny blob (azblob) package?

Reply:
Unfortunately, this will introduce a circular dependency as the SAS-parsing functionality in the `sas` package requires this type.  I think we'll need to move `ParseURL()` and its dependent types to `sas` and can type-alias them from other packages as needed.

Reply:
Is there another place we can put ParseURL to avoid the circular dep issue? It's not really a SAS thing at all.

Reply:
Yes, I think that's the better choice

Reply:
I suppose we just place all of this in `internal/exported` and type-alias the relevant parts.  But before I do that, can you take a look at this part of the `URLParts` definitnion.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/internal/exported/blob_url.go#L37-L38

This looks a little clunky to me in that we have one field for SAS-specific content (which is immutable no less, although the fields in the parent aren't) and an "unparsed" field for the other query parameters.

Reply:
This has been fixed.